Source/Particles/Resampling/VelocityCoincidenceThinning.cpp

```diff
@@ -28,15 +24,37 @@ VelocityCoincidenceThinning::VelocityCoincidenceThinning (const std::string& spe
         "Resampling min_ppc should be greater than or equal to 1"
     );
 
-    utils::parser::getWithParser(
-        pp_species_name, "resampling_algorithm_delta_ur", m_delta_ur
-    );
-    utils::parser::getWithParser(
-        pp_species_name, "resampling_algorithm_n_theta", m_ntheta
-    );
-    utils::parser::getWithParser(
-        pp_species_name, "resampling_algorithm_n_phi", m_nphi
+    std::string velocity_grid_type_str = "spherical";
+    pp_species_name.query(
+        "resampling_algorithm_velocity_grid_type", velocity_grid_type_str
     );
+    if (velocity_grid_type_str == "spherical") {
+        m_velocity_grid_type = VelocityGridType::Spherical;
+        utils::parser::getWithParser(
+            pp_species_name, "resampling_algorithm_delta_ur", m_delta_ur
+        );
+        utils::parser::getWithParser(
+            pp_species_name, "resampling_algorithm_n_theta", m_ntheta
+        );
+        utils::parser::getWithParser(
+            pp_species_name, "resampling_algorithm_n_phi", m_nphi
+        );
+    }
+    else if (velocity_grid_type_str == "cartesian") {
+        m_velocity_grid_type = VelocityGridType::Cartesian;
+        utils::parser::getWithParser(
```

Comment:
I was also thinking that these three could be combined into one input parameter, reducing the verbosity of the input file.

Reply:
I also considered that, but then I looked at our convention with similar parameters in other cases and it seems that we generally use an array for PICMI input but then specify each direction separately in the naive input.
Examples include:
- `GaussianBunchDistribution` which takes `rms_bunch_size` array as input but passes parameters to the native input (and same with `centroid_position`) as
```
        species.add_new_group_attr(source_name, 'x_rms', self.rms_bunch_size[0])
        species.add_new_group_attr(source_name, 'y_rms', self.rms_bunch_size[1])
        species.add_new_group_attr(source_name, 'z_rms', self.rms_bunch_size[2])
```
- `DensityDistributionBase` takes `lower_bound` and `upper_bound` arrays but passes
```
        species.add_new_group_attr(source_name, 'xmin', self.lower_bound[0])
        species.add_new_group_attr(source_name, 'xmax', self.upper_bound[0])
        species.add_new_group_attr(source_name, 'ymin', self.lower_bound[1])
        species.add_new_group_attr(source_name, 'ymax', self.upper_bound[1])
        species.add_new_group_attr(source_name, 'zmin', self.lower_bound[2])
        species.add_new_group_attr(source_name, 'zmax', self.upper_bound[2])
```
- Boundary potentials are also specified as separate input parameters for each boundary - `lo/hi_x/y/z`.

Of course there are counter examples like the specification of the domain lower and upper bounds and boundary condition types. So maybe we should decide on a single convention and start enforcing it for new code additions. Your preference is to pass arrays rather than individual direction parameters?

Reply:
I looked around in the code and noticed this pattern. Where there are multiple parameters like this, if they are all required, then they are input as a vector. Otherwise they are input separately. This makes sense since in the first case, putting them in a vector is compact and keeps closely coupled quantities together. In the latter, if they were input as a vector, a default or unset flag would be needed to specify the unset parameters in the vector.

I do agree that it would be good to have a standard here in how to specify related parameters like this.

Reply:
That makes sense, yes. Seeing as all three parameters are required in this case, I will switch to using an array as input.
Going forward I will also keep an eye out for this general rule.